src/coreclr/jit/lowerxarch.cpp

```diff
@@ -320,9 +320,192 @@ GenTree* Lowering::LowerBinaryArithmetic(GenTreeOp* binOp)
 
     ContainCheckBinary(binOp);
 
+#ifdef TARGET_AMD64
+    if (JitConfig.EnableApxConditionalChaining())
+    {
+        if (binOp->OperIs(GT_AND, GT_OR))
+        {
+            GenTree* next;
+            if (TryLowerAndOrToCCMP(binOp, &next))
+            {
+                return next;
+            }
+        }
+    }
+#endif // TARGET_AMD64
+
     return binOp->gtNext;
 }
 
+#ifdef TARGET_AMD64
+
+//------------------------------------------------------------------------
+// TryLowerAndOrToCCMP : Lower AND/OR of two conditions into test + CCMP + SETCC nodes.
+//
+// Arguments:
+//    tree - pointer to the node
+//    next - [out] Next node to lower if this function returns true
+//
+// Return Value:
+//    false if no changes were made
+//
+bool Lowering::TryLowerAndOrToCCMP(GenTreeOp* tree, GenTree** next)
```

Comment:
How much does this differ from the arm64 version? Can you just move that version to lower.cpp?

Reply:
I've gone through and lifted some common code up.

Reply:
I revered the code that lifts `TryLowerAndOrToCCMP` because I can see a number of changes were made.

Reply:
Those changes should be ok for xarch as well. I would still combine them.

Reply:
Ok, I've gone ahead and refactored the common code up.